### ISSUE TITLE
Bad checking for lostpassword hash

### ISSUE THREAD

**Issue by namaless** [woocommerce/woocommerce#12018]
## EXPLANATION OF THE ISSUE

After some hours I see the WooCommerce fail the lostpassword hash instead of WordPress working, I found the bug here: https://github.com/woocommerce/woocommerce/blob/master/includes/shortcodes/class-wc-shortcode-my-account.php#L333

After investigate more inside I see the problem is from the `$user->user_activation_key` used by wordpress use 2 keys then when I add the parser all working fine (take from wp-login.php):

```php
if ( false !== strpos( $user->user_activation_key, ':' ) ) {
    list( $pass_request_time, $pass_key ) = explode( ':', $user->user_activation_key, 2 );
} else {
    $pass_key = $user->user_activation_key;
}

if ( ! empty( $user ) ) {
	if ( empty( $wp_hasher ) ) {
		require_once ABSPATH . 'wp-includes/class-phpass.php';
		$wp_hasher = new PasswordHash( 8, true );
	}

	$valid = $wp_hasher->CheckPassword( $key, $pass_key );
}
```
## STEPS TO REPRODUCE THE ISSUE

Request a new password then change the url with lostpassword endpoint and you get all times the Invalid key message.

In another case you can test changed the `self::check_password_reset_key` method with the wordpress core `check_password_reset_key` and all working fine.

## SYSTEM STATUS REPORT

```
### WordPress Environment ###

Home URL: http://localhost
Site URL: http://localhost
WC Version: 2.6.4
Log Directory Writable: ❌ Per consentire i log
rendete /xxx/wp-content/uploads/wc-logs/ accessibile in scrittura o definite un valore di WC_LOG_DIR personalizzato.

WP Version: 4.6.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: it_IT

### Server Environment ###

Server Info: nginx/1.10.0
PHP Version: 7.0.8-0ubuntu0.16.04.2
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Il tuo server non ha la classe SoapClient attiva - alcuni plugin gateway che usano SOAP potrebbero non funzionare correttamente.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
POST remoto: ✔
GET remoto: ✔

### Database ###

WC Database Version: 2.6.4
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Active Plugins (17) ###

SB Welcome Email Editor: by Sean Barton – 4.8
AffiliateWP: by AffiliateWP LLC – 1.9.1
Akismet: by Automattic – 3.2
Payment Gateway Based Fees and Discounts for WooCommerce: by Algoritmika Ltd – 2.1.0
CodeMirror File Editor: by Viacheslav Zavoruev – 1.2.2
Contact Form 7: by Takayuki Miyoshi – 4.5.1
Cookie Law Bar: by WidgetPack  – 1.0
GetBowtied Tools: by GetBowtied – 1.2.1
Jetpack by WordPress.com: by Automattic – 4.3.1
WPBakery Visual Composer: by Michael M - WPBakery.com – 4.12.1
Mailgun: by Mailgun – 1.4
Members: by Justin Tadlock – 1.1.2
NunoPress Primo Label: by  –
Site Offline or Coming Soon: by wpecommerce – 1.6.6
User Switching: by John Blackbourn – 1.0.9
WooCommerce Checkout Manager: by Visser Labs – 4.0.9
WooCommerce: by WooThemes – 2.6.4

### Settings ###

Force SSL: –
Currency: EUR (€)
Currency Position: right
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2

### API ###

API Enabled: –

### WC Pages ###

Shop Base: ❌ La pagina non è impostata
Carrello: ❌ La pagina non è impostata
Cassa: #8 - /cassa/
Il mio account: ❌ La pagina non contiene lo shortcode: [woocommerce_my_account]

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: Primo Label Theme
Version: 1.0
Author URL: https://nunopress.com/
Child Theme: ✔
Parent Theme Name: Shopkeeper
Parent Theme Version: 1.7
Parent Theme Author URL: http://www.getbowtied.com/
WooCommerce Support: ✔

### Templates ###

Overrides: shopkeeper/woocommerce/archive-product.php
shopkeeper/woocommerce/cart/cart-empty.php
shopkeeper/woocommerce/checkout/form-billing.php
shopkeeper/woocommerce/checkout/form-checkout.php
shopkeeper/woocommerce/checkout/form-coupon.php
shopkeeper/woocommerce/checkout/form-login.php
shopkeeper/woocommerce/checkout/form-shipping.php
shopkeeper/woocommerce/checkout/payment-method.php
shopkeeper/woocommerce/checkout/thankyou.php
shopkeeper/woocommerce/content-product.php
shopkeeper/woocommerce/content-product_cat.php
shopkeeper/woocommerce/content-single-product.php
madeinitaly-app/woocommerce/emails/email-footer.php versione - il file non è aggiornato. La versione di base è 2.3.0
madeinitaly-app/woocommerce/emails/email-header.php versione - il file non è aggiornato. La versione di base è 2.4.0
shopkeeper/woocommerce/global/breadcrumb.php
shopkeeper/woocommerce/global/sidebar.php
shopkeeper/woocommerce/global/wrapper-end.php
shopkeeper/woocommerce/global/wrapper-start.php
shopkeeper/woocommerce/loop/loop-end.php
shopkeeper/woocommerce/loop/loop-start.php
shopkeeper/woocommerce/loop/no-products-found.php
shopkeeper/woocommerce/loop/orderby.php
shopkeeper/woocommerce/myaccount/form-login.php
shopkeeper/woocommerce/myaccount/my-downloads.php
shopkeeper/woocommerce/order/form-tracking.php
shopkeeper/woocommerce/single-product/add-to-cart/external.php
shopkeeper/woocommerce/single-product/add-to-cart/simple.php
shopkeeper/woocommerce/single-product/add-to-cart/variable.php
shopkeeper/woocommerce/single-product/meta.php
shopkeeper/woocommerce/single-product/product-image.php
shopkeeper/woocommerce/single-product/product-thumbnails.php
shopkeeper/woocommerce/single-product/rating.php
shopkeeper/woocommerce/single-product/related.php
shopkeeper/woocommerce/single-product/short-description.php
shopkeeper/woocommerce/single-product/tabs/additional-information.php
shopkeeper/woocommerce/single-product/tabs/description.php
shopkeeper/woocommerce/single-product/tabs/tabs.php
shopkeeper/woocommerce/single-product/up-sells.php
shopkeeper/woocommerce/single-product-reviews.php
shopkeeper/woocommerce/single-product.php

: Impara come aggiornare i template scaduti
```

**Comment by claudiosmweb** [comment]
> Request a new password then change the url with lostpassword endpoint and you get all times the Invalid key message.

So you are requesting the password change from the `wp-login.php` page and not from the `my-account/lost-password/`?

**Comment by namaless** [comment]
I tested from new registration notify, with woocommerce request new password working because use woocommerce activation key format and now used from wordpress `time:crypted_key`.

**Comment by claudiosmweb** [comment]
> I tested from new registration notify

What new registration notify?

**Comment by namaless** [comment]
Ok sorry if I explain badly.
- go in the admin / users / new / create new user with the checkbox for notify (send link to create new password)

Without any modification the link sended example: http://example.com/wp-login.php?action=rp&key=xxxx&login=example
If you change this link to: http://example.com/my-account/lost-password/?key=xxxx&login=example you get invalid key, but if you follow the wp link working fine. If you change the code insiede woocommerce (https://github.com/woocommerce/woocommerce/blob/master/includes/shortcodes/class-wc-shortcode-my-account.php#L202) with replace `check_password_reset_key` or use my code (get from wp-login.php) you see the woocommerce working fine.

**Comment by claudiosmweb** [comment]
Closed in favor of #12030
